### PR TITLE
Fix hydrator path overflow bug

### PR DESCRIPTION
**Pull request by StephenCathcart**
Fix for https://github.com/neo4j/neo4j-go-driver/issues/590. A new test case was added for regression.

### CODE REVIEW

File: neo4j/internal/bolt/hydrator_test.go [hunk 1]
```diff
@@ -1067,6 +1069,72 @@ func TestHydratorBolt5(outer *testing.T) {
...
+			}
+			// Indices
+			packer.ArrayHeader(c.size)
+			for i := 1; i <= c.size; i++ {
+				packer.Int(1)
+			}
+			// Test we can hydrate without a panic from int overflows.
+			buf, err := packer.End()
+			if err != nil {
+				panic("Build error")
```
Comment (fbiville) [inline neo4j/internal/bolt/hydrator_test.go]: nit: maybe `t.Fatal` instead of `panic`? (same below)

Comment (bigmontz) [reply neo4j/internal/bolt/hydrator_test.go]: What is the difference for the user experience?

Comment (bigmontz) [reply neo4j/internal/bolt/hydrator_test.go]: (now I see it is test code 🤦🏼 )

**Review by bigmontz — APPROVED** [review]
💯

File: neo4j/internal/bolt/hydrator_test.go (hunk 1 above)
Comment (StephenCathcart) [reply neo4j/internal/bolt/hydrator_test.go]: agreed thanks for the feedback 👍

**Pull request merged**